rules-reviewed/jboss-eap/jboss-eap5/jboss-eap5-java.windup.xml

```diff
@@ -0,0 +1,148 @@
+<?xml version="1.0"?>
+<ruleset id="jboss-eap5-java"
+         xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+
+    <!-- Rules converted from Tubame JBossRules.windup.xml -->
+
+    <metadata>
+        <description>
+            This ruleset provides analysis of JBoss EAP 5 .java files that may require individual attention
+            when migrating to JBoss EAP 6+.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,2.3.0.Final"/>
+            <addon id="org.jboss.windup.rules,windup-rules-java,2.3.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="eap" versionRange="[5,6)"/>
+        <targetTechnology id="eap" versionRange="[6,)"/>
+        <tag>jboss-eap5</tag>
+        <tag>jms</tag>
+        <tag>ejb</tag>
+        <tag>reviewed-2015-08-28</tag>
+    </metadata>
+
+    <rules>
+        <rule id="jboss-eap5-java-01000">
+            <when>
+                <javaclass references="org.jboss.ejb3.annotation.Management">
+                    <location>ANNOTATION</location>
+                </javaclass>
+            </when>
+            <perform>
+                <classification title="JBoss EAP 5 JMX ManagementBean" effort="3">
+                    <tag>jmx</tag>
+                </classification>
+            </perform>
+        </rule>
+        <rule id="jboss-eap5-java-02000">
+            <when>
+                <javaclass references="javax.jms.QueueConnectionFactory">
+                    <location>FIELD_DECLARATION</location>
+                    <location>VARIABLE_DECLARATION</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint effort="1" severity="MANDATORY">
+                    <message>`QueueConnectionFactory` was used to obtain connection to JMS queues.
+                        Replace the lookup string `QueueConnectionFactory` with `ConnectionFactory`.
+                    </message>
+                    <tag>jms</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="jboss-eap5-java-03000">
+            <when>
+                <javaclass references="javax.persistence.JoinColumn">
+                    <location>ANNOTATION</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint effort="3" severity="MANDATORY">
+                    <message>
+                        The `@JoinColumn` annotation specifies the column used by JPA provider for foreign keys.
```

Comment:
https://docs.oracle.com/javaee/5/api/javax/persistence/JoinColumn.html

I'm in favor of removing this rule. It doesn't seem to be correct at all as @JoinColumn is a standard Java EE API.